pilz_teleoperation/src/pilz_teleoperation/curses_key_input.py

```diff
@@ -75,9 +77,20 @@ def resolve_key_input(self):
         """ Read currently pressed key and publish it to the driver.
             Should be called with at least 10 HZ!
         """
-        key_code = self._screen.getch()
+        key_code = self._read_keyboard_input()
+
         if key_code in CursesKeyInput.MOVE_BINDINGS:
-            self._twist_pub.publish(Twist(linear=CursesKeyInput.MOVE_BINDINGS[key_code]))
+            self.__driver.set_twist_command(CursesKeyInput.MOVE_BINDINGS[key_code])
         elif key_code in CursesKeyInput.SETTING_BINDINGS:
-            self._setting_srv(SetTeleopSettingsRequest(pressed_commands=[CursesKeyInput.SETTING_BINDINGS[key_code]]))
+            new_settings = SetTeleopSettingsRequest(pressed_commands=[CursesKeyInput.SETTING_BINDINGS[key_code]])
+            self.__driver.set_teleop_settings(new_settings)
+
+    def _read_keyboard_input(self):
+        key_code = self._screen.getch()
+        self._clear_input_queue()
```

Comment:
```suggestion
        curses.flushinp()
```
There is a helper function to flush the buffer(s). But looking at the published messages now the repetition delay becomes obvious. The first press is published as speed, then zeros until repetition triggers continous velocity...